src/charm.py

```diff
@@ -150,76 +150,102 @@ def _get_existing_config_file(self, path: str) -> str:
             return existing_content_stringio.read()
         return ""
 
-    def _get_upf_host_port_list(self) -> List[Tuple[str, int]]:
+    def _get_upf_host_port_list(self, event: EventBase) -> List[Tuple[str, int]]:
         """Gets the list of UPF hosts and ports from the `fiveg_n4` relation data bag.
 
+        Args:
+            event (EventBase): Juju event that triggers the config update. Used to identify
+            the broken relation so that its information is not added to the config.
+
         Returns:
             List[Tuple[str, int]]: List of UPF hostnames and ports.
         """
         upf_host_port_list = []
+        broken_relation_id = event.relation.id if isinstance(event, RelationBrokenEvent) else None
         for fiveg_n4_relation in self.model.relations.get(FIVEG_N4_RELATION_NAME, []):
             if not fiveg_n4_relation.app:
                 logger.warning(
                     "Application missing from the %s relation data", FIVEG_N4_RELATION_NAME
                 )
-                return []
+                continue
+            if fiveg_n4_relation.id == broken_relation_id:
```

Comment:
The problem with this workaround is that it will hold until the next event. As far as I remember, when relation goes down, it's not being removed from the model. This means that when another event comes after the `relation_broken`, `self.model.relations.get(FIVEG_N4_RELATION_NAME, [])` will also return a broken relation, but this time we won't know that it's broken.
Early this week we discussed using `StoredState` as a potential workaround. @gatici is there any conclusion on this one?

Reply:
Hi,https://github.com/canonical/sdcore-amf-operator/pull/53/files#r1409330695

Reply:
In this PR, the situation is a bit different. self._configure_sdcore_nms is executed upon a relation broken event. Both self.model.relations and broken_relation_id are the parameters that may change quickly. I do not see a place to be kept in StoredState.

Reply:
Hi @benhoyt,
Could you comment on this? I implemented a workaround in which we identify a broken relation using the relation broken event. Like this, we skip the information in that databag since we no longer consider it valid.  As @Gmerold  stated, if another event triggers the config update just immediately, we would no be able to identify the relation broken anymore. 
Any idea how we could handle this situation ?

Reply:
I think this is again related to https://github.com/canonical/operator/issues/940, which we hope to address early next year. I believe once we fix that you won't need to determine "is the current event the relation-broken event for this relation ID" or pass the event around, but you'll be able to query the current state of the relations.

CC: @tonyandrewmeyer for visibility.

Reply:
@Gmerold do you know of a way to reproduce a relation being visible after relation-broken? @benhoyt & I discussed this with the Juju team today, and they felt that after the relation-broken event if you see the broken relation that would likely be a Juju error (possibly caused by using `--force`?).

In our planned solution for #940, you'll be able to identify the broken relation in the relation-broken event, and in any framework events (collect status, deferred events) that occur in the same Juju context. However, if there was a different event after relation-broken, during which Juju did list the broken relation in relation-ids, then this would indeed be a problem.

If you know how this could be reproduced, then we can try to do that, and figure out how to handle it and/or talk again to the Juju team. Thanks!